esphome/components/ld2410/number/__init__.py

```diff
@@ -0,0 +1,153 @@
+        and CONF_MAX_MOVE_DISTANCE_GATE in config
+        and CONF_MAX_STILL_DISTANCE_GATE in config
+    )
+    if has_some_timeout_configs and not has_all_timeout_configs:
+        raise cv.Invalid(
+            f"{CONF_TIMEOUT}, {CONF_MAX_MOVE_DISTANCE_GATE} and {CONF_MAX_STILL_DISTANCE_GATE} are all must be set"
+        )
+    for x in range(9):
+        move = CONF_MOVE_THRESHOLDS[x]
+        still = CONF_STILL_THRESHOLDS[x]
+        if (move in config or still in config) and (
+            move not in config or still not in config
+        ):
+            raise cv.Invalid(f"{move} and {still} are all must be set")
+    return config
+
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(CONF_LD2410_ID): cv.use_id(LD2410Component),
+        cv.Optional(CONF_TIMEOUT): number.number_schema(
+            MaxDistanceTimeoutNumber,
+            unit_of_measurement=UNIT_SECOND,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:clock-time-eight-outline",
+        ),
+        cv.Optional(CONF_MAX_MOVE_DISTANCE_GATE): number.number_schema(
+            MaxDistanceTimeoutNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:motion-sensor",
+        ),
+        cv.Optional(CONF_MAX_STILL_DISTANCE_GATE): number.number_schema(
+            MaxDistanceTimeoutNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:motion-sensor-off",
+        ),
+        cv.Optional(CONF_LIGHT_THRESHOLD): number.number_schema(
+            LightThresholdNumber,
+            device_class=DEVICE_CLASS_ILLUMINANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:car-light-high",
+        ),
+    }
+)
+
+for i in range(9):
+    CONFIG_SCHEMA = CONFIG_SCHEMA.extend(
+        cv.Schema(
+            {
+                cv.Optional(CONF_MOVE_THRESHOLDS[i]): number.number_schema(
+                    GateThresholdNumber,
+                    device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+                    unit_of_measurement=UNIT_PERCENT,
+                    entity_category=ENTITY_CATEGORY_CONFIG,
+                    icon="mdi:motion-sensor",
+                ),
+                cv.Optional(CONF_STILL_THRESHOLDS[i]): number.number_schema(
+                    GateThresholdNumber,
+                    device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+                    unit_of_measurement=UNIT_PERCENT,
+                    entity_category=ENTITY_CATEGORY_CONFIG,
+                    icon="mdi:motion-sensor-off",
+                ),
+            }
+        )
+    )
+
+CONFIG_SCHEMA = cv.All(CONFIG_SCHEMA, validate)
+
+
+async def to_code(config):
+    ld2410_component = await cg.get_variable(config[CONF_LD2410_ID])
+    if CONF_TIMEOUT in config:
+        n = await number.new_number(
+            config[CONF_TIMEOUT], min_value=0, max_value=65535, step=1
+        )
```

Comment:
See #5181
Please apply to all places that use a similar syntax
```suggestion
    if timeout_config := config.get(CONF_TIMEOUT):
        n = await number.new_number(
            timeout_config, min_value=0, max_value=65535, step=1
        )
```

Reply:
done in all relevant places